great_expectations/datasource/data_connector/configured_asset_s3_data_connector.py

```diff
@@ -85,8 +86,12 @@ def __init__(
 
         if boto3_options is None:
             boto3_options = {}
+        if credentials is None:
+            credentials = {}
 
         try:
+            # boto3_options takes priority over credentials in the case of duplicate keys
+            boto3_options = {**credentials, **boto3_options}
```

Comment:
My thought process here is that a user should be able to configure both the `credentials` and `boto3_options` keys within their YAML; if there is an overlap between the two due to duplicative contents, the `boto3_options` object takes precedence (as it is more specific).